src/rust-util/pelikan/src/protocol/admin.rs

```diff
@@ -13,11 +13,132 @@
 // limitations under the License.
 
 use super::*;
-use pelikan_sys::protocol::admin::{request, response};
+use ccommon::buf::OwnedBuf;
+use ccommon_sys::buf;
```

Comment:
This is probably outside of this review - but it seems a little weird to use both the wrapper and -sys crates here

Reply:
This ends up happening because this file is translating between a rust api and a C api - although it's really only needed since we need to cast pointers around.